### PR TITLE
Convert army system data to data model

### PR DESCRIPTION
**Pull request by CarlosFdez**

### CODE REVIEW

**Comment by simonward** [comment]
Would it be possible to add PublicationData to armies, or is that best served as a separate feature?

**Comment by CarlosFdez** [comment]
That would be a new feature. This is only a conversion to data models, on a document type that has received minimal migrations.

File: src/module/actor/army/data.ts [hunk 1]
```diff
@@ -1,115 +1,224 @@
...
+                ammunition: new fields.SchemaField({
+                    value: new fields.NumberField({ required: true, nullable: false, initial: 0 }),
+                }),
+                potions: new fields.SchemaField({
+                    value: new fields.NumberField({ required: true, nullable: false, initial: 0 }),
+                }),
+            }),
+            traits: new fields.SchemaField({
+                value: new fields.ArrayField(new fields.StringField({ required: true, nullable: false })),
+                rarity: new fields.StringField({ required: true, nullable: false, initial: "common" }),
```
Comment (stwlam) [inline src/module/actor/army/data.ts:90]: needs choices

Comment (CarlosFdez) [reply src/module/actor/army/data.ts]: Done, though I didn't test it.

File: src/module/actor/army/data.ts [hunk 2]
```diff
@@ -1,115 +1,229 @@
...
+                ranged: new fields.SchemaField(createWeaponSchema(), { required: true, nullable: true, initial: null }),
+            }),
+            resources: new fields.SchemaField({
+                /** How often this army can use ranged attacks */
+                ammunition: new fields.SchemaField({
+                    value: new fields.NumberField({ required: true, nullable: false, initial: 0 }),
+                }),
+                potions: new fields.SchemaField({
+                    value: new fields.NumberField({ required: true, nullable: false, initial: 0 }),
+                }),
```
Comment (stwlam) [inline src/module/actor/army/data.ts:86]: integer, min 0?

File: src/module/actor/army/data.ts [hunk 3]
```diff
@@ -1,115 +1,229 @@
...
+            }),
+            consumption: new fields.NumberField({ required: true, nullable: false, initial: 1 }),
+            scouting: new fields.NumberField({
+                required: true,
+                nullable: false,
+                initial: (d: DeepPartial<ArmySystemSource>) =>
+                    getLevelDefault(ARMY_STATS.scouting, d.details?.level?.value),
+            }),
+            recruitmentDC: new fields.NumberField({ required: true, nullable: false, initial: 15 }),
```
Comment (stwlam) [inline src/module/actor/army/data.ts:60]: integer?

File: src/module/actor/army/data.ts [hunk 4]
```diff
@@ -1,115 +1,229 @@
...
+        return {
+            ...parent,
+            ac: new fields.SchemaField({
+                value: new fields.NumberField({
+                    required: true,
+                    nullable: false,
+                    initial: (d: DeepPartial<ArmySystemSource>) =>
+                        getLevelDefault(ARMY_STATS.ac, d.details?.level?.value),
+                }),
+                potency: new fields.NumberField({ required: true, nullable: false, initial: 0 }),
```
Comment (stwlam) [inline src/module/actor/army/data.ts:38]: integer?

File: src/module/actor/army/data.ts [hunk 5]
```diff
@@ -1,115 +1,229 @@
...
+import fields = foundry.data.fields;
+
+class ArmySystemData extends ActorSystemModel<ArmyPF2e, ArmySystemSchema> {
+    static override defineSchema(): ArmySystemSchema {
+        const parent = super.defineSchema();
+
+        function createWeaponSchema(): ArmyWeaponSchema {
+            return {
+                name: new fields.StringField({ required: true, nullable: false }),
+                potency: new fields.NumberField({ required: true, nullable: false, initial: 0 }),
```
Comment (stwlam) [inline src/module/actor/army/data.ts:21]: integer, min 0?

Comment (stwlam) [reply src/module/actor/army/data.ts]: No initial required

Comment (stwlam) [reply src/module/actor/army/data.ts]: (but the initial has to meet the min, of course)

File: src/module/actor/army/data.ts [hunk 6]
```diff
@@ -1,115 +1,238 @@
...
+            ...parent,
+            ac: new fields.SchemaField({
+                value: new fields.NumberField({
+                    required: true,
+                    nullable: false,
+                    integer: true,
+                    initial: (d: DeepPartial<ArmySystemSource>) =>
+                        getLevelDefault(ARMY_STATS.ac, d.details?.level?.value),
+                }),
+                potency: new fields.NumberField({ required: true, nullable: false, integer: true, min: 0 }),
```
Comment (stwlam) [inline src/module/actor/army/data.ts:39]: I think this might need an explicit initial since the default is `null`

**Pull request merged**